docs/reference/rubicon-objc-api.rst

```diff
@@ -63,17 +63,14 @@ Objective-C classes
 
         The method's return type, as a :mod:`ctypes` type, or ``None`` for ``void`` methods.
 
-    .. attribute:: argtypes
+    .. attribute:: imp_argtypes
```

Comment:
No argument that these changes are needed; however, we should be wary about changing interfaces, especially when they're documented. 

We're definitely "pre-1.0", so from that perspective I'm not *overly* concerned - getting the API *right* is more important than preserving strict backwards compatibility. However, we should probably start thinking about which of these attributes we consider to be part of the public API that we're committing to maintaining, and which are useful internals that we're documenting because they're useful, but we reserve the right to change if they get in the way.

Reply:
Right, I thought about that too. In the case of `ObjCMethod` I don't think it's a big issue, because (quoting the docs):

> Currently, Rubicon doesn't provide any documented way to get an unbound `ObjCMethod` object for an instance method of an `ObjCClass`.

That is, the only documented/supported way to get an `ObjCMethod` object is by creating one directly from an Objective-C `Method` pointer. This isn't something that many people are likely to do, and [a quick GitHub search](https://github.com/search?q=rubicon+objcmethod&type=Code) shows no external uses of `ObjCMethod`.

If this was after the release of 1.0, I would add a property with the old name for compatibility, and have it raise a `DeprecationWarning` and redirect to the new property. But as we're still before 1.0 and this feature probably has no external users, I think it's okay to just rename the property with no compatibility handling.

Reply:
On second thought, perhaps it would be a good idea to "un-document" `ObjCMethod` completely for now, as there's already almost no good way to use it. That would give us more freedom to work on its API, until we add proper introspection features to allow users to access a class's methods and such.